cmd/deploy/deploy.go

```diff
@@ -226,12 +227,36 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, opts *Option
 					buildErrs = append(buildErrs, err.Error())
 					continue
 				}
+
+				if okteto.IsOkteto() && registry.IsOktetoRegistry(opts.Tag) {
```

Comment:
We should expand envars for all type of registries, no?

Reply:
Also, I think we should prepend the pipeline name to the image name, to avoid collisions with same build names in a different pipeline

Reply:
> We should expand envars for all type of registries, no?
you mean expand or set? i thought we could only access the registry when is the okteto registry, so the envs are related to the okteto registries? 

> prepend the pipeline name to the image name

you mean the name param in the manifest? i was looking into this today to change it when we create the image field when parsing the manifest, but this field is optional, so in case there is no name at the manifest, what would be?

Reply:
it's extracted from the repo name. @ifbyol could you link the function that extract the pipeline name?

Reply:
We use this function https://github.com/okteto/okteto/blob/master/cmd/utils/manifest.go#L24 to get the name from the repo URL or the current folder (if it is not a repository)